### PR TITLE
Remove root detailed guide paths

### PR DESCRIPTION
**Pull request by binaryberry**
Detailed guides are now available under guidance/ and moving
away from root paths.

This is part of our multi-step [plan](https://docs.google.com/a/digital.cabinet-office.gov.uk/spreadsheets/d/1xMgOH3ha7dzoPaHJw9VAu9hlCvAm3ifwbBABbuFNMDQ/edit?usp=sharing) to achieve this.

### CODE REVIEW

File: db/migrate/20150805124936_remove_root_detailed_guides_paths.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class RemoveRootDetailedGuidesPaths < Mongoid::Migration
+  def self.up
+    Artefact.where(kind: 'detailed_guide').each do |detailed_guide|
+      unless detailed_guide.path.start_with?(%r{^/guidance/})
+        begin
+          old_path = detailed_guide.path
+          detailed_guide.set(:paths, [])
```
Comment (boffbowsh) [inline db/migrate/20150805124936_remove_root_detailed_guides_paths.rb]: `detailed_guide.set(:paths, ["/guidance/#{detailed_guide.slug}"])`

File: db/migrate/20150805124936_remove_root_detailed_guides_paths.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class RemoveRootDetailedGuidesPaths < Mongoid::Migration
+  def self.up
+    Artefact.where(kind: 'detailed_guide').each do |detailed_guide|
+      unless detailed_guide.path.start_with?(%r{^/guidance/})
```
Comment (boffbowsh) [inline db/migrate/20150805124936_remove_root_detailed_guides_paths.rb]: `s/path/slug`

Comment (boffbowsh) [inline db/migrate/20150805124936_remove_root_detailed_guides_paths.rb]: Actually delete this unless condition, do it for all.

File: db/migrate/20150805124936_remove_root_detailed_guides_paths.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class RemoveRootDetailedGuidesPaths < Mongoid::Migration
+  def self.up
+    Artefact.where(kind: 'detailed_guide').each do |detailed_guide|
+      begin
+        detailed_guide.set(:paths, ["/#{detailed_guide.slug}"])
+        detailed_guide.save!
+        puts "deleted #{old_path} path"
```
Comment (boffbowsh) [inline db/migrate/20150805124936_remove_root_detailed_guides_paths.rb]: This variable is no longer set

File: db/migrate/20150805124936_remove_root_detailed_guides_paths.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+class RemoveRootDetailedGuidesPaths < Mongoid::Migration
+  def self.up
+    Artefact.where(kind: 'detailed_guide').each do |detailed_guide|
+      begin
+        detailed_guide.set(:paths, ["/#{detailed_guide.slug}"])
+        detailed_guide.save!
```
Comment (boffbowsh) [inline db/migrate/20150805124936_remove_root_detailed_guides_paths.rb]: If you're calling `set`, `save` isn't reuqired.

**Comment by boffbowsh** [comment]
This should be changed to be a rake task in the same way as https://github.com/alphagov/panopticon/pull/282.

**Comment by binaryberry** [comment]
Count of both this rake task and the reslug rake task are off. Can't figure out why.

screen shot 2015-08-06 at 17 47 25

**Comment by boffbowsh** [comment]
I've pushed a fix, it's due to "undocumented behaviour" (I'm being kind) in Mongoid.

**Pull request merged**